1.train_models/log_reg_plates_cp_fs_data/plate_4/nbconverted/plate_4_classify_genotype.py

```diff
@@ -0,0 +1,248 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# # Train logistic regressions to classify genotypes for plate 4
+
+# In[1]:
+
+
+import pathlib
+import warnings
+
+import numpy as np
+import pandas as pd
+from joblib import dump
+from scipy.stats import uniform
+from sklearn.exceptions import ConvergenceWarning
+from sklearn.linear_model import LogisticRegression
+from sklearn.model_selection import RandomizedSearchCV, train_test_split
+from sklearn.preprocessing import LabelEncoder
+from sklearn.utils import parallel_backend
+
+# ## Find the root of the git repo on the host system
+
+# In[2]:
+
+
+# Get the current working directory
+cwd = pathlib.Path.cwd()
+
+if (cwd / ".git").is_dir():
+    root_dir = cwd
+
+else:
+    root_dir = None
+    for parent in cwd.parents:
+        if (parent / ".git").is_dir():
+            root_dir = parent
+            break
+
+# Check if a Git root directory was found
+if root_dir is None:
+    raise FileNotFoundError("No Git root directory found.")
+
+
+# ## Define and create paths
+
+# In[3]:
+
+
+# Input paths
+plate_path = "Plate_4_sc_norm_fs.parquet"
+plate_path = pathlib.Path(f"{root_dir}/nf1_painting_repo/3.processing_features/data/feature_selected_data/{plate_path}")
+
+# Output paths
+models_path = pathlib.Path("trained_models")
+data_path = pathlib.Path("model_data")
+
+# Create output paths if nonexistent
+models_path.mkdir(parents=True, exist_ok=True)
+data_path.mkdir(parents=True, exist_ok=True)
+
+
+# ## Load the dataframe
+
+# In[4]:
+
+
+platedf = pd.read_parquet(plate_path)
+
+
+# ## Define column names
+
+# In[5]:
+
+
+well_column = "Metadata_Well"
+gene_column = "Metadata_genotype"
+concentration_column = "Metadata_Concentration"
+
+
+# # Process data
+
+# ## Down-sample by genotype
+
+# In[6]:
+
+
+# Filter data where the siRNA construct concentration is zero
+platedf = platedf[platedf[concentration_column] == 0]
+
+min_gene = platedf[gene_column].value_counts().min()
+platedf = platedf.groupby(gene_column, group_keys=False).apply(lambda x: x.sample(n=min_gene, random_state=0))
+
+
+# ## Encode genotypes
+
+# In[7]:
+
+
+# Encode classes
+le = LabelEncoder()
+featdf = platedf.copy()
+featdf[gene_column] = le.fit_transform(featdf[gene_column])
+
+
+# ## Remove Metadata and encode labels
+
+# In[8]:
+
+
+# Create dataframe without Metadata
+feat_columns = [col for col in platedf.columns if "Metadata" not in col]
+featdf = platedf[feat_columns]
+
+# Encode labels
+platedf[gene_column] = le.transform(platedf[gene_column])
+
+
+# ## Split the Data
+
+# In[9]:
+
+
+test_frac = 0.2
+val_frac = 0.15
+
+X_train, X_test, y_train, y_test = train_test_split(featdf, platedf[gene_column], test_size = test_frac, random_state=0, shuffle=True, stratify=platedf[gene_column])
```

Comment:
Consider making the `shuffle=True` arg modular so you can run this for both unshuffled and shuffled data

Reply:
We talked about this one